src/java.base/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -670,6 +686,55 @@ public void interpret(Deque<Object> stack, StoreFunc storeFunc,
         }
     }
 
+    /**
+     * ShiftLeft([shiftAmount])
+     *   Shifts the Bytes on the top of the operand stack (unsigned).
+     *   Positive [shiftAmount] converts to long if needed and shifts left.
+     *   Negative [shiftAmount] shifts right and converts to int if needed.
+     */
+    record ShiftLeft(int shiftAmount, Class<?> type) implements Binding {
```

Comment:
Please split this into 2 binding operators: one for ShiftLeft and another for (arithmetic) ShiftRight, rather than mixing the 2 implementations into a single operator.

For the right shift you could then also use a positive shift amount, and avoid the double negation that's needed right now.

Reply:
Done.